.github/workflows/test.yml

```
@@ -32,19 +33,7 @@ jobs:
       # Checkout the repository
       - name: Checkout repository
         uses: actions/checkout@v4
-        
-      # Produce a valid 'settings.xml' file for Maven by
-      # running a script to inject the environment variables
-      # provided by GitHub secrets
-      - name: Set up Maven credentials
-        working-directory: ./.github/scripts
-        env:
-          MAVEN_USERNAME: ${{github.actor}}
-          MAVEN_PASSWORD: ${{github.token}}
-        run: |
-          chmod +x set-maven-credentials.sh
-          ./set-maven-credentials.sh ..
-          
+
```

```suggestion
permissions:
  contents: read

```

add secrets.GITHUB_TOKEN's permissions here

---

I agreed that it doesn't do any harm to set this here although we should probably set the default permissions to read-only rather than read-write.